src/renderers/dom/fiber/wrappers/ReactDOMFiberSelect.js

```diff
@@ -148,20 +148,22 @@ var ReactDOMSelect = {
       wasMultiple: !!props.multiple,
     };
 
-    if (
-      props.value !== undefined &&
-      props.defaultValue !== undefined &&
-      !didWarnValueDefaultValue
-    ) {
-      warning(
-        false,
-        'Select elements must be either controlled or uncontrolled ' +
-          '(specify either the value prop, or the defaultValue prop, but not ' +
-          'both). Decide between using a controlled or uncontrolled select ' +
-          'element and remove one of these props. More info: ' +
-          'https://fb.me/react-controlled-components',
-      );
-      didWarnValueDefaultValue = true;
+    if (__DEV__) {
+      if (
+        props.value !== undefined &&
+        props.defaultValue !== undefined &&
+        !didWarnValueDefaultValue
```

Comment:
We should also move the `var didWarnValueDefaultValue = ...` declaration behind the DEV block.